ScoutBot/Services/DatabaseService.cs

```diff
@@ -10,27 +10,30 @@ namespace ScoutBot.Services
     public class DatabaseService
     {
         /// <summary>
-        /// Adds the providerId to the database.
+        /// Adds new SheetAccess row to the table.
         /// </summary>
-        /// <param name="providerId">The id from riot for generating tournaments.</param>
-        public static async Task PutProviderAsync(int providerId)
+        /// <param name="sheetId">The google sheet id.</param>
+        /// <param name="roleId">The discord role id.</param>
+        /// <returns></returns>
+        public static async Task<bool> AddSheetAccess(string sheetId, string roleId)
         {
-            using (StatsContext db = new StatsContext())
+            using (ScoutContext db = new ScoutContext())
             {
-                await db.Providers.AddAsync(new Provider { ProviderID = providerId });
-                db.SaveChanges();
-            }
-        }
-
-        /// <summary>
-        /// Sees if there is already an entry in the providers table.
-        /// </summary>
-        public static async Task<bool> ProviderExistsAsync()
-        {
-            using (StatsContext db = new StatsContext())
-            {
-                return await db.Providers.AnyAsync();
+                try
+                {
+                    await db.SheetAccess.AddAsync(new SheetAccess
+                    {
+                        SheetId = sheetId,
+                        RoleId = roleId
+                    });
+                    db.SaveChanges();
+                }
+                catch
+                {
+                    return false;
```

Comment:
This call should never fail, I'm not a fan of tossing the error away here and just returning `false` instead. You may be missing out on some important error message.

Reply:
It can fail, for instance it will throw an exception if a the roleId is not unique.